### PR TITLE
Update `starknet_getBlock*` methods to match OpenRPC specification 1.0.0-rc1

### PR DESCRIPTION
**Pull request by kkovaacs**
This PR updates our JSON-RPC API to match this part of the latest specification. Changes include:

- Updated BLOCK type: `sequencer` is renamed to `sequencer_address`, `accepted_time` is now `timestamp`, and `old_root` and `gas_price` have been removed.
- Remove `starknet_getBlockByHash` and `starknet_getBlockByNumber` methods.
- Add `starknet_getBlockWith{TxHashes,Txs}` methods.
- Remove `TransactionAndReceipt` RPC type.

Note that internally we're still using the same storage backend implementation as before, but `BlockResponseScope` is now not exposed on the API. This means that querying for transaction hashes is still suboptimal, since the underlying implementation loads full transaction data and then throws everything but the hashes away...

### CODE REVIEW

**Review by Mirko-von-Leipzig — APPROVED** [review]

**Review by CHr15F0x — APPROVED** [review]

**Review by Mirko-von-Leipzig — APPROVED** [review]

**Review by CHr15F0x — APPROVED** [review]

**Comment by koivunej** [comment]
Converting to a draft as this is meant to follow-up some other PR.

File: crates/pathfinder/src/rpc/api.rs [hunk 1]
```diff
@@ -59,6 +59,19 @@ pub struct RawBlock {
...
+pub enum BlockResponseScope {
+    TransactionHashes,
+    FullTransactions,
+}
+
+impl Default for BlockResponseScope {
+    fn default() -> Self {
+        BlockResponseScope::TransactionHashes
+    }
+}
```
Comment (Mirko-von-Leipzig) [inline crates/pathfinder/src/rpc/api.rs:73]: Why does this need a default? And why is it `TransactionHashes`?

**Review by Mirko-von-Leipzig — APPROVED** [review]
LGTM

File: crates/pathfinder/rpc_examples.sh [hunk 2]
```diff
@@ -11,28 +11,14 @@ function rpc_call() {
...
 
-rpc_call '{"jsonrpc":"2.0","id":"0","method":"starknet_getBlockByHash","params":["pending"]}'
-rpc_call '{"jsonrpc":"2.0","id":"1","method":"starknet_getBlockByNumber","params":["pending"]}'
+rpc_call '{"jsonrpc":"2.0","id":"0","method":"starknet_getBlockWithTxs","params":["pending"]}'
+rpc_call '{"jsonrpc":"2.0","id":"2","method":"starknet_getBlockWithTxHashes","params":["pending"]}'
 
-rpc_call '{"jsonrpc":"2.0","id":"2","method":"starknet_getBlockByHash","params":["pending","TXN_HASH"]}'
-rpc_call '{"jsonrpc":"2.0","id":"3","method":"starknet_getBlockByNumber","params":["pending","TXN_HASH"]}'
+rpc_call '{"jsonrpc":"2.0","id":"4","method":"starknet_getBlockWithTxs","params":["latest"]}'
+rpc_call '{"jsonrpc":"2.0","id":"6","method":"starknet_getBlockWithTxHashes","params":["latest"]}'
```
[the remark this answers is not in the document]
Comment (kkovaacs) [reply crates/pathfinder/rpc_examples.sh:18]: We have an example for that on line 21. Or what do you mean?

[the remark this answers is not in the document]
Comment (Mirko-von-Leipzig) [reply crates/pathfinder/rpc_examples.sh:18]: ID numbers are arbitrary and not important.

**Review by CHr15F0x — APPROVED** [review]
LGTM, I appreciate the `check_result` helper

**Pull request merged**